### PR TITLE
Feature/better image popup design

### PR DESCRIPTION
**Pull request by Sonusiety**

### CODE REVIEW

File: app/src/main/java/com/github/sdpsharelook/ImagePopupFragment.kt [hunk 1]
```diff
@@ -45,11 +49,18 @@ class ImagePopupFragment : DialogFragment() {
...
+        view.findViewById<TextView>(R.id.popUpDate2).text = Date(arguments?.getLong(KEY_DATE)!!).toString().dropLast(14)
+        view.findViewById<Button>(R.id.button3).setOnClickListener { dismiss() }
     }
+
+//    private fun showBottomSheetDialog() {
+//        val bottomSheetDialog = BottomSheetDialog(requireActivity())
+//        bottomSheetDialog.setContentView(R.layout.image_popup_layout)
+//        bottomSheetDialog.findViewById<TextView>(R.id.popUpSource2).text
+//        bottomSheetDialog.show()
+//    }
```
Comment (villemontc) [inline app/src/main/java/com/github/sdpsharelook/ImagePopupFragment.kt:65]: Comments

File: app/src/main/java/com/github/sdpsharelook/MapsFragment.kt [hunk 2]
```diff
@@ -120,12 +125,17 @@ open class MapsFragmentLift : Fragment(R.layout.fragment_maps) {
...
+                imageView.load(word.picture!!)
+            }
             if (word != null) {
                 ImagePopupFragment.newInstance(
                     word.source.toString(),
                     word.target.toString(),
                     word.savedDate!!,
-                    decodeImage(word.picture!!)
+                    imageView.drawable.toBitmap()
+                    //decodeImage(word.picture!!)
```
Comment (villemontc) [inline app/src/main/java/com/github/sdpsharelook/MapsFragment.kt:138]: Comments

**Review by villemontc — APPROVED** [review]

File: app/src/main/java/com/github/sdpsharelook/ImagePopupFragment.kt [hunk 3]
```diff
@@ -45,11 +49,18 @@ class ImagePopupFragment : DialogFragment() {
...
+        view.findViewById<ImageView>(R.id.popUpImage2).setImageBitmap(arguments?.getParcelable(
             KEY_IMAGE))
-        view.findViewById<TextView>(R.id.popUpSource).text = arguments?.getString(KEY_SOURCE)
-        view.findViewById<TextView>(R.id.popUpTarget).text = arguments?.getString(KEY_TARGET)
-        view.findViewById<TextView>(R.id.popUpDate).text = Date(arguments?.getLong(KEY_DATE)!!).toString()
-        view.findViewById<Button>(R.id.button).setOnClickListener { dismiss() }
+        view.findViewById<TextView>(R.id.popUpSource2).text = arguments?.getString(KEY_SOURCE)
+        view.findViewById<TextView>(R.id.popUpTarget2).text = arguments?.getString(KEY_TARGET)
+        view.findViewById<TextView>(R.id.popUpDate2).text = Date(arguments?.getLong(KEY_DATE)!!).toString().dropLast(14)
+        view.findViewById<Button>(R.id.button3).setOnClickListener { dismiss() }
```
Comment (rombirli) [inline app/src/main/java/com/github/sdpsharelook/ImagePopupFragment.kt:57]: No need to rename popupImage -> popupImage2

Comment (rombirli) [reply app/src/main/java/com/github/sdpsharelook/ImagePopupFragment.kt:57]: you just have to name it popupImage in image_popup_layout

File: app/src/main/java/com/github/sdpsharelook/MapsFragment.kt [hunk 4]
```diff
@@ -120,12 +125,17 @@ open class MapsFragmentLift : Fragment(R.layout.fragment_maps) {
         map.setOnMarkerClickListener { marker ->
             val word = markerMap[marker]
             BitmapFactory.decodeResource(requireContext().resources, R.drawable.default_user_path)
+            val imageView : ImageView = ImageView(requireContext())
+            if (word != null) {
```
Comment (rombirli) [inline app/src/main/java/com/github/sdpsharelook/MapsFragment.kt:129]: word?.let{}

File: app/src/main/res/layout/timeline_layout.xml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        <TextView
+            android:id="@+id/textView"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text=" Timeline"
+            android:textSize="24sp"
+            android:textStyle="bold" />
+
+        <TextView
+            android:id="@+id/textView2"
```
Comment (rombirli) [inline app/src/main/res/layout/timeline_layout.xml:28]: bad id

File: app/src/main/res/layout/timeline_layout.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    ]]>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
+
+        <TextView
+            android:id="@+id/textView"
```
Comment (rombirli) [inline app/src/main/res/layout/timeline_layout.xml:20]: bad id

File: app/src/main/res/layout/timeline_layout.xml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+        <TextView
+            android:id="@+id/textView2"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="Location"
+            android:textAlignment="center" />
+
+        <TextView
+            android:id="@+id/textView3"
```
Comment (rombirli) [inline app/src/main/res/layout/timeline_layout.xml:35]: bad id

File: app/src/main/res/layout/timeline_layout.xml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="110dp"
+            android:text="Description"
+            android:textAlignment="center" />
+
+        <SeekBar
+            android:id="@+id/seekBar"
```
Comment (rombirli) [inline app/src/main/res/layout/timeline_layout.xml:42]: bad id

File: app/src/main/res/layout/timeline_layout.xml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            android:id="@+id/textView2"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="Location"
+            android:textAlignment="center" />
+
+        <TextView
+            android:id="@+id/textView3"
+            android:layout_width="match_parent"
+            android:layout_height="110dp"
```
Comment (rombirli) [inline app/src/main/res/layout/timeline_layout.xml:37]: Why do we need hardcoded height here ?

File: app/src/main/res/layout/modal_bottom_sheet_dialog.xml [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+            android:layout_gravity="center_vertical"
+            android:padding="8dp"/>
+
+    </LinearLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:id="@+id/uploadLinearLaySout"
```
Comment (rombirli) [inline app/src/main/res/layout/modal_bottom_sheet_dialog.xml:56]: typo in word uploadLinearLaySout (you mean uploadLinearLayout ?)

File: app/src/main/res/layout/modal_bottom_sheet_dialog.xml [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:id="@+id/download"
```
Comment (rombirli) [inline app/src/main/res/layout/modal_bottom_sheet_dialog.xml:11]: try to keep coherency in names. Either rename shareLinearLayout -> share and uploadLinearLayout -> upload or rename download -> downloadLinearLayout

**Review by rombirli — APPROVED** [review]

**Pull request merged**